lib/Mojolicious/Guides/Cookbook.pod

```diff
@@ -65,6 +65,76 @@ connections concurrently.
 
 In L<Mojolicious> this event loop is L<Mojo::IOLoop>.
 
+=head2 Reverse Proxy
+
+A reverse proxy architecture is a deployment technique used in many production
+environments, where a I<reverse proxy> server is put in front of your
+application to act as the endpoint accessible by external clients. It can
+provide a lot of benefits, like terminating SSL connections from the outside,
+limiting the number of concurrent open sockets towards the Mojolicious
+application (or even using Unix sockets), balancing load across multiple
+instances, or supporting several applications through the same IP/port.
+
+   .               ..........................................
+                   :                                        :                        
+                   :                       +-------------+  :                        
+                   :  +------------+      +-------------+|  :
+   +--------+      :  |            |      |             ||  :
+   |        |-------->|  reverse   |----->| Mojolicious ||  :
+   | client |      :  |   proxy    |      | application ||  :
+   |        |<--------|            |<-----|             ||  :
+   +--------+      :  | (frontend) |      |  (backend)  ||  :
+                   :  |            |      |             |+  :
+                   :  +------------+      +-------------+   :                       
+                   :                                        :
+                   .. system boundary (e.g. same host) ......
+
+This setup introduces some problems, though: the application will receive
+requests from the reverse proxy instead of the original client; the
+address/hostname where your application lives internally will be different from
+the one visible from the outside; if terminating SSL, the reverse proxy exposes
+services via HTTPS, while using HTTP towards the Mojolicious application.
+
+As an example, compare a sample request from the client and what the Mojolicious
+application receives:
+
+   client                       reverse proxy                Mojolicious app
+    __|__              _______________|______________             ____|____
+   /     \            /                              \           /         \
+   1.2.3.4 --HTTPS--> api.example.com      10.20.30.39 --HTTP--> 10.20.30.40
+
+   GET /foo/1 HTTP/1.1                |    GET /foo/1 HTTP/1.1
+   Host: api.example.com              |    Host: 10.20.30.40
+   User-Agent: Mojolicious (Perl)     |    User-Agent: ShinyProxy/1.2
+   ...                                |    ...
+
+The client address is no longer available (it might be useful for analytics, or
```

Comment:
However, now the client address is no longer available (which might be useful for analytics or